packages/wallet-sdk/src/sign/SignerConfigurator.ts

```diff
@@ -26,11 +33,11 @@ export class SignerConfigurator {
   private popupCommunicator: PopUpCommunicator;
   private updateListener: SignRequestHandlerListener;
 
-  private signerTypeStorage = new ScopedLocalStorage('CBWSDK', 'SignerConfigurator');
-  private signerTypeSelectionResolver: ((signerType: string) => void) | undefined;
+  private injectedSigner: Signer | undefined;
+  private initiatedWLSigner: WLSigner | undefined;
 
-  signerType: string | null;
-  signer: Signer | undefined;
+  private signerTypeStorage = new ScopedLocalStorage('CBWSDK', 'SignerConfigurator');
+  private resolveSelectSignerType?: (_: SignerType) => void;
```

Comment:
can we remove this?
seems it does only two things
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/1a92dfb4-dd88-429a-9c87-7eb4ffc38991)

- store: it could be directly called when `this.resolveSelectSignerType?.()` is being called
- resolve when it's done: doesn't seem like it's observed by anyone?

Reply:
i dont think we can tho, this resolver is being called at 2 places
1. SCW&Extension: at handleSignerConfigMessage() when receiving SignerTypeSelected
2. WalletLink: at L159, when WLAdaptor update status

Reply:
what do they await for? why do we have a separate resolver?

Reply:
it's the same resolver for selecting a signerType, but we call it at different place

1. SCW and extension is considered selected, when user click the button from SCW popup
2. WalletLink is considered selected after WalletLink Adaptor established connection

Reply:
sounds like we just need common interface for selection updates on all those signer types?

Reply:
hmm wdym by common interface
i tried simplified this area a bit:
1. broke down this resolver to only take the resolve
2. created a private function to handle signerTypeSelected